osu.Game/Overlays/Settings/Sections/UserInterface/MainMenuSettings.cs

```diff
@@ -43,5 +52,17 @@ private void load(OsuConfigManager config)
                 }
             };
         }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            backgroundSourceDropdown.Current.BindValueChanged(source =>
```

Comment:
Shouldn't this invoke on changes to `user` instead, along with taking the value change event and check supporter with the new value?

Reply:
Yeah... I was thinking of another implementation along the way (adding a `Default` option), then checking that source and supporter. Could be useful for users who always want the default background, and a UX improvement for non-supporters as they now know what the background source is overridden to (change note to: "This setting was overridden to default as it is only for osu!supporters.").

We can check for both the user and source if that's the right way forward.

Reply:
Sounds like a good improvement for sure